key-concepts/covenants.adoc

```diff
@@ -1,11 +1,174 @@
 = Covenants
 
-A covenant is a set of link:actions.adoc[action] creators and types, a
-reducer (which is a link:smart_contracts.adoc[smart contract], and a
-Redux link:sagas.adoc[saga] that together indicate the functioning of an
-Interbit blockchain. These actions, smart contract, and saga work
-together to fulfill the functionality of a single Interbit blockchain.
-The `rootSaga` in a smart contract is optional.
+[quote,Merriam Webster Dictionary]
+**covenant**: a written agreement or promise, usually under seal,
+between two or more parties, especially for the performance of some
+action.
+
+Interbit blockchain networks are governed by covenants. Each node in the
+blockchain's network agrees to apply the blockchain's covenant.
+
+image:img/covenant-chains.svg["Each node in an Interbit blockchain
+network applies the same covenant", 500, 170, role="zoom center"]
+
+The covenant dictates how the blockchain can be changed over time. As a
+result, all participating nodes in the blockchain's network process
+actions and update the blockchain's state in a deterministic fashion.
+
+
+== Anatomy
+
+image:img/covenant-anatomy.svg["The anatomy of an Interbit covenant",
+522, 220, align="center" role="zoom center"]
+
+An Interbit covenant is comprised of several components:
+
+. A set of link:actions.adoc[action] creators and types. Any changes to
+  be made to a blockchain begin with an action creator that creates an
+  action of a type known to the blockchain.
+
+. A link[smart_contracts.adoc[smart contract]. An Interbit smart
+  contract is a block of Javascript code that processes each action
+  dispatched to the blockchain.
+
+. A link:sagas.adoc[saga]. A saga is used to coordinate any
+  non-deterministic operations such that only deterministic changes are
+  made to the blockchain.
```

Comment:
The initial state, which provides a shared starting point for all nodes to process from deterministically.